src/Microsoft.SqlTools.ServiceLayer/Connection/ReliableConnection/ReliableConnectionHelper.cs

```diff
@@ -460,6 +463,56 @@ private static bool IsCloudEngineId(int engineEditionId)
         }
 
         /// <summary>
+        /// Determines if the type of database that a connection is being made to is SQL data warehouse.
+        /// </summary>
+        /// <param name="connection"></param>
+        /// <returns>True if the database is a SQL data warehouse</returns>
+        public static bool IsSqlDwDatabase(IDbConnection connection)
+        {
+            Validate.IsNotNull(nameof(connection), connection);
+
+            Func<string, bool> executeCommand = commandText =>
+            {
+                bool result = false;
+                ExecuteReader(connection,
+                              commandText,
+                              readResult: (reader) =>
+                              {
+                                  reader.Read();
+                                  int engineEditionId = int.Parse(reader[0].ToString(), CultureInfo.InvariantCulture);
+
+                                  result = IsSqlDwEngineId(engineEditionId);
+                              }
+                    );
+                return result;
+            };
+
+            bool isSqlDw = false;
+            try
+            {
+                isSqlDw = executeCommand(SqlConnectionHelperScripts.EngineEdition);
+            }
+            catch (SqlException)
+            {
+                // The default query contains a WITH (NOLOCK).  This doesn't work for Azure DW, so when things don't work out, 
+                // we'll fall back to a version without NOLOCK and try again.
+                isSqlDw = executeCommand(SqlConnectionHelperScripts.EngineEditionWithLock);
```

Comment:
> isSqlDw = executeCommand(SqlConnectionHelperScripts.EngineEditionWithLock); [](start = 15, length = 76)

should this be inside a try/catch too?  or is someone else catch exceptions thrown from this call?